yt/yt/library/containers/cri/cri_executor.cpp

```diff
@@ -351,6 +352,9 @@ class TCriExecutor
         {
             auto* metadata = config->mutable_metadata();
             metadata->set_name(ctSpec->Name);
+
+            // Set unique attempt for each newly created container.
+            metadata->set_attempt(Attempt_++);
```

Comment:
What this attempt field actually mean in CRI?

For me attempt notion is usually used in retries. But here we have CriExecutor that is responsible for all requests to CRI

Reply:
"Attempt" is designed to be used for retries for sure.
But it's just a number appened to container name.

https://github.com/containerd/containerd/blob/main/pkg/cri/server/helpers.go#L128

Unique key for create is (namespace, pod, name, attempt)
it returns container id and all futher operations are done by id.

We could add randomness to container name with same result.
At the end of each job CRI executor list sand removes all contianers in the pod thus garbage should be collected.

Reply:
I would prefer to add randomness to the name, because the current solution is misleading.